bot.py

```diff
@@ -35,8 +36,373 @@ async def challenge(ctx: commands.Context, opponent: str):
+					}, {
+						"ID": 322,
+						"Name": "Cosmic Power",
+						"Type": 8192,
+						"Category": 0,
+						"Targets": 7,
+						"CurrentPP": 20,
+						"MaxPP": 20,
+						"Priority": 0,
+						"Power": 0,
+						"Accuracy": 0
+					}, {
+						"ID": 280,
+						"Name": "Brick Break",
+						"Type": 2,
+						"Category": 1,
+						"Targets": 10,
+						"CurrentPP": 15,
+						"MaxPP": 15,
+						"Priority": 0,
+						"Power": 75,
+						"Accuracy": 100
+					}],
+					"Friendship": 0,
+					"OriginalTrainerID": 0,
+					"Type": 0
+				}
+			}, {
+				"Team": 1,
+				"Party": 1,
+				"Slot": 0,
+				"Pokemon": {
+					"Name": "Absol",
+					"NatDex": 359,
+					"Level": 60,
+					"TotalExperience": 211060,
+					"Gender": 0,
+					"IVs": [4, 2, 3, 3, 1, 1],
+					"EVs": [5, 31, 11, 41, 29, 40],
+					"Nature": 2,
+					"Stats": [151, 166, 80, 102, 81, 101],
+					"StatModifiers": [0, 0, 0, 0, 0, 0, 0, 0, 0],
+					"StatusEffects": 0,
+					"CurrentHP": 151,
+					"Moves": [{
+						"ID": 230,
+						"Name": "Sweet Scent",
+						"Type": 1,
+						"Category": 0,
+						"Targets": 11,
+						"CurrentPP": 20,
+						"MaxPP": 20,
+						"Priority": 0,
+						"Power": 0,
+						"Accuracy": 100
+					}, {
+						"ID": 429,
+						"Name": "Mirror Shot",
+						"Type": 256,
+						"Category": 2,
+						"Targets": 10,
+						"CurrentPP": 10,
+						"MaxPP": 10,
+						"Priority": 0,
+						"Power": 65,
+						"Accuracy": 85
+					}, {
+						"ID": 284,
+						"Name": "Eruption",
+						"Type": 512,
+						"Category": 2,
+						"Targets": 11,
+						"CurrentPP": 5,
+						"MaxPP": 5,
+						"Priority": 0,
+						"Power": 150,
+						"Accuracy": 100
+					}, {
+						"ID": 157,
+						"Name": "Rock Slide",
+						"Type": 32,
+						"Category": 1,
+						"Targets": 11,
+						"CurrentPP": 10,
+						"MaxPP": 10,
+						"Priority": 0,
+						"Power": 75,
+						"Accuracy": 90
+					}],
+					"Friendship": 0,
+					"OriginalTrainerID": 0,
+					"Type": 0
+				}
+			}]
+		})
+	await ctx.send(f"Turn: {type(turn)} {turn.toJSON()}")
+
+RESPONSE_REACTIONS = ["🇦", "🇧", "🇨", "🇩"]
+
+async def prompt_for_turn(user: discord.User, battlecontext) -> Turn:
+	# TODO: create an actual class for battlecontext instead of just parsing the json into a dict like a monkey
+	# TODO: prompt user for what type of turn
+
+	if not user.dm_channel:
+		await user.create_dm()
+	embed = discord.Embed(title=battlecontext["Pokemon"]["Name"])
+	for move in battlecontext["Pokemon"]["Moves"]:
+		embed.add_field(name=move['Name'], value=f"{move['Type']} {move['CurrentPP']} {move['MaxPP']}")
+	msg: Message = await user.dm_channel.send(content="Select a move", embed=embed)
+	for i in RESPONSE_REACTIONS:
+		await msg.add_reaction(i)
+	def check(payload):
+		return payload.message_id == msg.id and payload.user_id == user.id and str(payload.emoji) in RESPONSE_REACTIONS
+	try:
+		# HACK: reaction_add doesn't work in DMs
+		payload = await bot.wait_for("raw_reaction_add", timeout=10, check=check)
```

Comment:
```suggestion
		payload = await bot.wait_for("raw_reaction_add", timeout=20, check=check)
```

Reply:
@dyc3 It timed out too fast in my opinion; by the time I got to my dm's the timeout had hit

Reply:
whoops meant to remove this